### PR TITLE
ASoC: SOF: remove unnecessary header files including

### LINKED ISSUE #357: upstream: revisit required header files
most of the includes are not necessary, need to reduce the list to the minimum required. it's not good practice to copy/paste across unrelated files.

### PR DESCRIPTION
**Pull request by keyonjie**
This is aimed to fix #357 

Only keep header files we do need included.

Signed-off-by: Keyon Jie <yang.jie@linux.intel.com>

### CODE REVIEW

**Comment by ranj063** [comment]
@keyonjie wow. this is a nice cleanup! let me test it to make sure nothing complains.

File: include/sound/sof.h [hunk 1]
```diff
@@ -11,11 +11,6 @@
 #ifndef __INCLUDE_SOUND_SOF_H
 #define __INCLUDE_SOUND_SOF_H
 
-#include <linux/kernel.h>
-#include <linux/types.h>
-#include <linux/interrupt.h>
-#include <linux/device.h>
-#include <linux/platform_device.h>
 #include <linux/pci.h>
```
Comment (plbossart) [inline include/sound/sof.h:14]: do we actually need pci.h as a baseline? Can be be added, e.g. in hda.h and sof-pci-dev?

**Comment by xiulipan** [comment]
@plbossart 
There is no 0-day now.
Should we include more ARCH for the CI build?
We only run build for some kconfig on X86_64

**Comment by plbossart** [comment]
> @plbossart
> There is no 0-day now.
> Should we include more ARCH for the CI build?
> We only run build for some kconfig on X86_64

I am not asking the CI to be based on 0-day, I am asking if we can force 0-day to run wild on this PR.
I just received a notice of a broken build with some randconfigs, removing 0-day completely is not in our best interest.

**Comment by keyonjie** [comment]
yes, I spent hours on figuring out this yesterday(it made me fever :-( ?), try and make sure every line removing don't block the compiling with my kernel config that all SOF platforms I known are selected.

we need to make sure it won't block any random config, hope 0-day can help here.

@ranj063 it's strange those files you mentioned are all passed in my kernel building.

File: include/sound/sof.h (hunk 1 above)
Comment (keyonjie) [reply include/sound/sof.h:14]: I believe we don't. we actually use it in several files(e.g. all those using struct pci_dev), putting pci.h including here is only a lazy mode for us as most of them will include sof.h

**Comment by xiulipan** [comment]
@mengdonglin 
Can we let 0-day back with some manually trigger?

**Comment by plbossart** [comment]
> yes, I spent hours on figuring out this yesterday(it made me fever :-( ?), try and make sure every line removing don't block the compiling with my kernel config that all SOF platforms I known are selected.
> 
> we need to make sure it won't block any random config, hope 0-day can help here.

Yes. I am not saying your work is bad, but that some platforms are really odd includes and we want to compile for all platforms even if SOF does not support them. It's useful since it catches other problems, just another tool we can use to improve code quality.
> 
> @ranj063 it's strange those files you mentioned are all passed in my kernel building.

Probably because changes since 4.14?

**Comment by plbossart** [comment]
Anyways it needs to be rebased due to conflicts.

**Comment by wenqingfu** [comment]
> @mengdonglin
> Can we let 0-day back with some manually trigger?

I'm afraid 0-day based pre-merge CI will bring you more pain to maintain it though extra good on this one. If the post-merge 0-day can detect problems, as long as we respond to it in time, that's good enough. After all, that's what other 0-day monitored repositories do - aftermath reaction.

**Comment by plbossart** [comment]
On 12/5/18 3:31 PM, Michael Fu wrote:
>
>     @mengdonglin <https://github.com/mengdonglin>
>     Can we let 0-day back with some manually trigger?
>
> I'm afraid 0-day based pre-merge CI will bring you more pain to 
> maintain it though extra good on this one. If the post-merge 0-day can 
> detect problems, as long as we respond to it in time, that's good 
> enough. After all, that's what other 0-day monitored repositories do - 
> aftermath reaction.
>
Agree, I am only asking for offline results for platforms not covered by 
the existing CI builds. We shouldn't gate integration if a build fails 
on an exotic platform, it's more of the icing on the cake before 
upstream contributions we are after.
>

**Comment by xiulipan** [comment]
@plbossart @wenqingfu 
Or we can just port the random config generator from 0-day and manually trigger some PR test with keyword?

**Comment by keyonjie** [comment]
> > yes, I spent hours on figuring out this yesterday(it made me fever :-( ?), try and make sure every line removing don't block the compiling with my kernel config that all SOF platforms I known are selected.
> > we need to make sure it won't block any random config, hope 0-day can help here.
> 
> Yes. I am not saying your work is bad, but that some platforms are really odd includes and we want to compile for all platforms even if SOF does not support them. It's useful since it catches other problems, just another tool we can use to improve code quality.
> 
> > @ranj063 it's strange those files you mentioned are all passed in my kernel building.
> 
> Probably because changes since 4.14?

@plbossart yes, I just traced and found that on 2018-06-27 Takashi add '#include <linux/pm_runtime.h>' into hdaudio.h which we include in our sof-priv.h.

That can explain why we don't have it on 4.14.

Thanks @ranj063 for reporting that, let me add it back and update PR to fix it for 4.14.

**Comment by keyonjie** [comment]
@plbossart updated with glk compiling issue fixed, please help review.

**Pull request merged**